### PR TITLE
Remove default parameter of ShufflerIterDataPipe

### PR DESCRIPTION
**Pull request by NicolasHug**
Closes https://github.com/pytorch/data/issues/298. This PR:

- removes the `default` parameter of `ShufflerIterDataPipe`
- renames `set_shuffle_setting()` into `set_shuffle()`
- let `set_shuffle()` return `self`.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #74370

Differential Revision: [D35073666](https://our.internmc.facebook.com/intern/diff/D35073666)

### CODE REVIEW

File: torch/utils/data/datapipes/iter/combinatorics.py [hunk 1]
```diff
@@ -104,8 +103,9 @@ def buffer_replace(buffer, x):
         buffer[idx] = x
         return val
 
-    def set_shuffle_settings(self, shuffle=True):
+    def set_shuffle(self, shuffle=True):
```
Comment (NicolasHug) [inline torch/utils/data/datapipes/iter/combinatorics.py:106]: Another suggestion I have for this would be `enable_shuffle()` or `enable_shuffling()`. But no strong opinion.

File: torch/utils/data/graph_settings.py [hunk 2]
```diff
@@ -31,5 +31,5 @@ def apply_shuffle_settings(datapipe, shuffle):
         graph = torch.utils.data.graph.traverse(datapipe, only_datapipe=True)
         all_pipes = get_all_graph_pipes(graph)
         for pipe in all_pipes:
-            if hasattr(pipe, 'set_shuffle_settings'):
-                pipe.set_shuffle_settings(shuffle)
+            if hasattr(pipe, 'set_shuffle'):
```
Comment (pmeier) [inline torch/utils/data/graph_settings.py:34]: Shouldn't we do `isinstance(pipe, Shuffler)` here? Checking for a public attribute seems sketchy.

File: torch/utils/data/datapipes/iter/combinatorics.py [hunk 3]
```diff
@@ -95,7 +94,7 @@ def __init__(self,
         else:
             self.datapipe = datapipe.unbatch(unbatch_level=unbatch_level)
         self.buffer_size = buffer_size
-        self._shuffle_enabled = default
+        self._shuffle_enabled = True
```
Comment (pmeier) [inline torch/utils/data/datapipes/iter/combinatorics.py:97]: Maybe just
```suggestion
        self._enabled = True
```

?

**Review by pmeier — APPROVED** [review]
Patch LGTM, thanks Nicolas! I'm a little confused that CI is not tripping? Are there no tests for this? cc @ejguan @NivekT

File: test/test_datapipe.py [hunk 4]
```diff
@@ -1577,6 +1577,10 @@ def test_shuffler_mapdatapipe(self):
         shuffler_dp = dp.map.Shuffler(input_dp2, indices=['a', 'b', 'c', 'd', 'e'])
         self.assertEqual(set(range(1, 6)), set(shuffler_dp))
 
+        #  Functional Test: deactivate shuffling via set_shuffle
+        shuffler_dp = dp.iter.Shuffler(input_dp1).set_shuffle(False)
+        self.assertEqual(list(shuffler_dp), list(input_dp1))
```
Comment (NicolasHug) [inline test/test_datapipe.py:1582]: I just realized this test function is about `dp.map.Shuffler` (which I did not know existed), but the one we're interested in is  `dp.iter.Shuffler`. LMK where it would make more sense to have this test?

Comment (NivekT) [reply test/test_datapipe.py:1582]: Here:

https://github.com/pytorch/pytorch/blob/a8e81ccad8c22e805c84a11f661f7cb643f606dd/test/test_datapipe.py#L1384

**Comment by NivekT** [comment]
> Patch LGTM, thanks Nicolas! I'm a little confused that CI is not tripping? Are there no tests for this? cc @ejguan @NivekT

It is related to the new unit test being in the `MapDataPipe` version rather than `IterDataPipe` version. See my other comment.

**Review by ejguan — APPROVED** [review]
Thanks for fixing it. LGTM with the similar comment with Philip

**Comment by NicolasHug** [comment]
@NicolasHug has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35073666).

**Review by VitalyFedyunin — APPROVED** [review]
